aiida/tools/dbimporters/plugins/matp.py

```diff
@@ -0,0 +1,217 @@
+
+    def _verify_api_key(self):
+        """
+        Verify the supplied API key by issuing a request to Materials Project.
+        """
+        response = requests.get(
+            'https://www.materialsproject.org/rest/v1/api_check', headers={'X-API-KEY': self._api_key})
+        response_content = response.json()  # a dict
+        if 'error' in response_content:
+            raise RuntimeError(response_content['error'])
+        if not response_content['valid_response']:
+            raise RuntimeError('Materials Project did not give a valid response for the API key check.')
+        if not response_content['api_key_valid']:
+            raise RuntimeError('Your API key for Materials Project is not valid.')
+
+    @property
+    def api_key(self):
+        """
+        Return the API key configured for the importer
+        """
+        return self._api_key
+
+    @property
+    def collection(self):
+        """
+        Return the collection that will be queried
+        """
+        return self._collection
+
+    @property
+    def get_supported_keywords(self):
+        """
+        Returns the list of all supported query keywords
+
+        :return: list of strings
+        """
+        return self._supported_keywords
+
+    def query(self, **kwargs):
+        """
+        Query the database with a given dictionary of query parameters for a given collection
+
+        :param query: a dictionary with the query parameters
+        :param collection: the collection to query
+        """
+        try:
+            query = kwargs['query']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `query` as a key. This '
+                                 'should contain a dictionary with the right query needed.')
+        try:
+            collection = kwargs['collection']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `collection` as a key.')
+
+        if not isinstance(query, dict):
+            raise TypeError('The query argument should be a dictionary')
+
+        if collection is None:
+            collection = self._collection
+
+        if collection == 'structure':
+            results = []
+            collection_list = ['structure', 'material_id', 'cif']
+            for entry in self.find(query, collection_list):
+                results.append(entry)
+            search_results = MatProjSearchResults(results, return_class=MatProjCifEntry)
+        else:
+            raise ValueError('Unsupported collection: {}'.format(collection))
+
+        return search_results
+
+    def find(self, query, collection):
+        """
+        Query the database with a given dictionary of query parameters
+
+        :param query: a dictionary with the query parameters
+        """
+        for entry in self._mpr.query(criteria=query, properties=collection):
+            yield entry
+
+
+class MatProjEntry(DbEntry):
```

Comment:
Keyworded argument ``license`` might be upstreamed to the baseclass, ``DbEntry``, and the ``MatProjEntry`` and ``MatProjEntry`` could be removed altogether, as it is required only to support the ``license`` argument.

Reply:
Yes, I just followed that was in the MPDS importer. Fixed.